integration/app/test-demo/README.md

```diff
@@ -0,0 +1,81 @@
+
+Different tests tend to use different setups. Creating a separate test application for each setup would not make sense and would add additional maintenance work. Following the example of [WebRTC samples](https://webrtc.github.io/samples/), we come up with the concept of a Mini Test App - a minimalist, single-page application containing only the necessary components for a single test. All Mini Test Apps are sub-pages of the React Testing Demo.
+
+## Guideline
+
+When adding tests for a functionality, build a new Mini Test App.
+
+* For example, we already have a "Roster Test App" and you want to add tests for the video feature. You should create a "Video Test App".
+
+When adding tests for new feature of an existing functionality, improve the existing Mini Test App.
+
+* For example, we already have a "Roster Test App" and you want to add tests for the new hook which returns the size of the roster. You should improve the "Roster Test App" to cover this hook.
+
+## Run the React Testing Demo Locally
+
+1. If you haven't already, [configure your AWS credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html).
+2. Install the dependencies: `npm install`.
+3. Start the webpack server and node server concurrently: `npm start`.
+4. Open <http://localhost:9000/> in your browser.
+
+## Add a New Mini Test App
+
+1. Implement a new Mini Test App under `pages` folder.
+   * You could reuse the components and containers to facilitate your development.
+
+2. Create a new route for Mini Test App in `src/app.tsx`.
+
+    ```js
+    const App: React.FC = () => (
+      <ThemeProvider theme={lightTheme}>
+        <Router>
+          <Routes>
+            <Route path={routes.HOME} element={<Home />} />
+            <Route path={routes.ROSTER_TEST_APP} element={<RosterTestApp />} />
+            {/* Add Route to new Mini Test App below */}
+          </Routes>
+        </Router >
+      </ThemeProvider>
+    );
+
+    export const routes = {
+      HOME: '/',
+      ROSTER_TEST_APP: 'roster-test',
+      // Add new routes below
+    };
+    ```
+
+3. Add the new Mini Test App to `pages/Home.tsx`.
+   * Add the test coverage of Mini Test App following the example.
+
+    ```tsx
+    export const Home: React.FC = () => (
+      <>
+        <Heading level={4}>
+          React Component Library Test Demo
+        </Heading>
+        <nav>
+          <ul>
+            <li>
+              <Link id="roster-test" to={routes.ROSTER_TEST_APP}>Roster Test App</Link>
+              <header>Component</header>
+              <ul>
+                <li>RosterAttendee</li>
+              </ul>
+              <header>Hook</header>
+              <ul>
+                <li>useRosterState</li>
+                <li>useAttendeeStatus</li>
+                <li>useAttendeeAudioStatus</li>
+                <li>useAudioVideo</li>
+              </ul>
+            </li>
+            {/* Add new links to Mini Test App below */}
+          </ul>
```

Comment:
Can we make this into a re-usable component? Maybe call it like `TestAppList` or something? 

```
<TestAppList
    header="RosterTestApp"
    components={arrayOfRosterComponents}
    hooks={arrayOfHooksUsed}
</>
```

Reply:
Updated:
```tsx
export const Home: React.FC = () => (
  <>
    <Heading level={4}>React Component Library Test Demo</Heading>
    <TestApps>
      <TestApp
        name="Roster Test"
        route={routes.ROSTER_TEST}
        components={["RosterAttendee"]}
        hooks={[
          "useRosterState",
          "useAttendeeStatus",
          "useAttendeeAudioStatus",
          "useAudioVideo",
        ]}
      />
      {/* Add new links to Mini Test App below */}
    </TestApps>
  </>
);
```